packages/cli/src/server/http-server.ts

```diff
@@ -17,7 +26,58 @@ export class HttpServer {
     });
   }
 
+  public redirectResponse(): Promise<string> {
+    return new Promise((resolve) => {
+      const interval = setInterval(() => {
+        if (this.token) {
+          clearInterval(interval);
+          resolve(this.token);
+        }
+      }, 300);
+    });
+  }
+
   public close(): void {
     this.server.close();
   }
+
+  private handleRedirectRequest(req: http.IncomingMessage) {
+    this.token = new URLSearchParams(req.url.slice(REDIRECT_ROUTE.length)).get('token');
+  }
+
+  private async handleWidgetDemo(res: http.ServerResponse): Promise<void> {
+    return new Promise((resolve) => {
+      const dashboardPath = path.resolve(__dirname, '../constants/dashboard/index.html');
+
+      fs.readFile(dashboardPath, 'utf8', (error, content) => {
+        if (error) {
+          if (error.code === 'ENOENT') {
+            res.end('404', 'utf-8');
+
+            return resolve();
```

Comment:
I think that it might be better to reject instead of resolve? So if we need we can catch errors later. Wdyt?

Reply:
I totally agree.